packages/sdk-js/src/sticky-bucket-service.ts

```diff
@@ -89,6 +89,32 @@ export abstract class StickyBucketService {
   }
 }
 
+export abstract class StickyBucketServiceSync extends StickyBucketService {
+  abstract getAssignmentsSync(
+    attributeName: string,
+    attributeValue: string
+  ): StickyAssignmentsDocument | null;
+
+  abstract saveAssignmentsSync(doc: StickyAssignmentsDocument): void;
+
+  getAllAssignmentsSync(
+    attributes: Record<string, string>
+  ): Record<StickyAttributeKey, StickyAssignmentsDocument> {
+    const docs: Record<string, StickyAssignmentsDocument> = {};
+    Object.entries(attributes)
+      .map(([attributeName, attributeValue]) =>
+        this.getAssignmentsSync(attributeName, attributeValue)
+      )
+      .forEach((doc) => {
+        if (doc) {
+          const key = `${doc.attributeName}||${doc.attributeValue}`;
```

Comment:
We have this same delimited string formatting duplicated now 7 times (4 in this file, 3 in core.ts).  We should probably abstract it to a function.  

```ts
export function getStickyBucketKey(attributeName: string, value: string) {
  return `${attributeName}||${value}`;
}
```